### PR TITLE
Allow parsing of yml swagger files in rswag-api

### PR DESCRIPTION
**Pull request by bclewis1**
Hello,
This project is really nice, I enjoyed writing specs using the dsl and everything felt very natural.
We are currently monkey patching our project as below because we have a large existing collection of swagger yamls.

### CODE REVIEW

File: rswag-api/lib/rswag/api/middleware.rb [hunk 1]
```diff
@@ -29,6 +30,18 @@ def call(env)
 
       private
 
+      def parse_file(filename)
+        if /\.yml$/ === filename
```
Comment (jaydorsey) [inline rswag-api/lib/rswag/api/middleware.rb]: Two suggestions: 

- `filename.end_with?('.yml')` (if you only want .yml)
- `/\.ya?ml$/ === filename` (the [yaml faq](https://yaml.org/faq.html) suggests .yaml, so might be nice to support it)

**Comment by jaydorsey** [comment]
I like this as an idea. Even the swagger tool online prompts you to convert your json to yaml when you paste it.  Plus, most of the examples on the swagger site are in yaml.

File: rswag-api/lib/rswag/api/middleware.rb (hunk 1 above)
Comment (bclewis1) [reply rswag-api/lib/rswag/api/middleware.rb]: :+1:, sorry about the delay fixing this

**Comment by BookOfGreg** [comment]
I just spent a couple of days learning how Rack serves files statically without needing to deal with mimetypes without any risk of path traversal issues:
https://github.com/rack/rack/blob/master/lib/rack/static.rb

It's quite interesting.

The takeaway from this is I'll merge this PR (yay! YAML support) and you may see me working on this set of files in the near future.
Also if you like please look into how static file serving appears to be done by others as I feel that I need certainty that this solution doesn't put anyone at risk.

Thank you for your contributions!

**Pull request merged**